python/mxnet/contrib/quantization.py

```diff
@@ -499,6 +499,9 @@ def quantize_model(sym, arg_params, aux_params,
     if quantized_dtype not in ('int8', 'uint8'):
         raise ValueError('unknown quantized_dtype %s received,'
                          ' expected `int8` or `uint8`' % quantized_dtype)
+    if quantized_dtype == 'uint8' and ctx != cpu():
+        raise ValueError('currently, uint8 quantization is only supported by CPU,'
+                         ' please switch to the context of CPU or int8 data type for GPU')
```

Comment:
better to add this error to backend like in the case for MKLDNN with int8 so that we dont have to add to other frontends when we support quantization.

Reply:
currently, only python frontend support quantization and in fact calibration progress will not use backend specific quantized operator. So I think it's good to add error message in this place currently.

Reply:
In `QuantizeCompute` (`quantize-inl.h`) you can check if std::is_same<xpu,gpu>::value and check for param.out_type and throw exception.

Reply:
I don't think this modification can work since **infer type** error `mxnet.base.MXNetError: [02:07:55] /home/ubuntu/experimentals/1.4_release/src/operator/quantization/../tensor/matrix_op-inl.h:250: Check failed: src.type_flag_ == ret.type_flag_ (3 vs. 5)` will occur before `QuantizeCompute` and we cannot get the ctx information during `infer` stage. So I think it's good to interrupt this action during the calibration stage.

Reply:
isnt that called from the forward pass of quantized_conv ? The quantize forward pass should execute before this.

Reply:
add `check src_type` in `quantized_conv.cu`,  please take a review again.